### PR TITLE
OOP school library: set up associations

### PR DESCRIPTION
**Pull request by akos29**
### OOP school library: set up associations

In this request majorly:

- [ ] add `Classroom` class 
- [ ] add `has-many/ belongs to` relationship between `Classroom` and `Student`
- [ ] add `Rental` class
- [ ] add `Book` class
- [ ] add `many-to-many` relationship between `Person` and `Book`

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class Book
+  attr_accessor :title, :author
+  attr_reader :rentals
```
Comment (ShadyShawkat) [inline book.rb]: - [x] Kindly convert the `rentals` to `attr_accessor` as you already use it for writing in the Rental class initialize method.

File: book.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+
+  def initialize(title:, author:)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+end
```
Comment (ShadyShawkat) [inline book.rb]: - [x] Please make sure to define the `add_rental` method that describes the many-to-many relation between the `books` and `persons` classes.

File: person.rb [hunk 3]
```diff
@@ -5,14 +5,15 @@ def correct_name
 end
 
 class Person < Nameable
-  attr_reader :id
+  attr_reader :id, :rentals
   attr_accessor :name, :age
```
Comment (ShadyShawkat) [inline person.rb]: - [x] Kindly convert the `rentals` to `attr_accessor` as you already use it for writing in the Rental class initialize method.

File: person.rb [hunk 4]
```diff
@@ -5,14 +5,15 @@ def correct_name
...
 
   def initialize(age, name = 'Unknown', parent_permission: true)
     @id = Random.rand(1..1000)
     @name = name
     @age = age
     @parent_permission = parent_permission
+    @rentals = []
     super()
   end
```
Comment (ShadyShawkat) [inline person.rb]: - [x] Please make sure to define the `add_rental` method that describes the many-to-many relation between the `books` and `persons` classes.

**Review by ShadyShawkat — CHANGES_REQUESTED** [review]
## Hi @akos29, 

Good job so far! 👍🏻 
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight
- All linters are passing ✔️ 
- Correct use of GitFlow ✔️ 
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 

Image

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by AmaduKamara — APPROVED** [review]
# Status: Approved 🟢 🟢 🟢

Hi @akos29, 

Good job working on the changes highlighted by the previous reviewer 👍🏽

Your project is complete!  There is nothing else to say other than... it's time to merge it: ship it:
Congratulations! 🎉 

## Highlights:

✅ You used Gitflow correctly 👍🏽
✅ Linter checks passed 👍🏽
✅ Set up associations correctly 👍🏽
✅ Good commit messages 👍🏽
✅ Good descriptive PR 👍🏽

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**